### PR TITLE
Add the option to build with Qt 6

### PR DESCRIPTION
**Pull request by cjmayo**
I've tried to do this in the way that makes it easiest to remove Qt 5 down the line.
For now there is no gamepad support with Qt 6.

### CODE REVIEW

**Comment by Neumann-A** [comment]
You probably want to avoid the whole code duplication and rather have a variable called `OPENSCAD_QT_VERSION` which can be either set to 5 or 6 (or auto). See e.g. https://gitlab.kitware.com/vtk/vtk/-/blob/master/CMake/vtkQt.cmake?ref_type=heads

File: CMakeLists.txt [hunk 1]
```diff
@@ -513,11 +516,20 @@ if(NOT HEADLESS)
...
+    if (APPLE AND EXISTS /opt/homebrew/opt/qt@6)
+      list(APPEND CMAKE_PREFIX_PATH "/opt/homebrew/opt/qt@6")
+    endif()
+  else()
+    if (APPLE AND EXISTS /usr/local/opt/qt@5)
+      list(APPEND CMAKE_PREFIX_PATH "/usr/local/opt/qt@5")
+    endif()
+    if (APPLE AND EXISTS /opt/homebrew/opt/qt@5)
+      list(APPEND CMAKE_PREFIX_PATH "/opt/homebrew/opt/qt@5")
+    endif()
```
Comment (kintel) [inline CMakeLists.txt:532]: This is getting a bit messy - do you think there's a way of moving this out of the main CMakeLists file?

Same goes for the Qt5 vs. Qt6 logic. Perhaps a separate macro which deals with the details without making the main CMakeLists file so messy?

**Comment by kintel** [comment]
FYI: I managed to build this on macOS - looks good so far

**Comment by kintel** [comment]
@cjmayo If you rebase this to master, it should build on macOS too.

File: CMakeLists.txt (hunk 1 above)
Comment (kintel) [reply CMakeLists.txt:533]: ok, let's defer this until we have both Qt5+6 working and have a better picture

**Comment by kintel** [comment]
We can probably merge this soon. I'll do a more proper review, and update #4909 accordingly

**Review by kintel — APPROVED** [review]

**Pull request merged**